pkg/chunk/storage/caching_index_client.go

```diff
@@ -69,8 +71,14 @@ func (s *cachingIndexClient) Stop() {
 }
 
 func (s *cachingIndexClient) QueryPages(ctx context.Context, queries []chunk.IndexQuery, callback func(chunk.IndexQuery, chunk.ReadBatch) (shouldContinue bool)) error {
-	// We cache the entire row, so filter client side.
-	callback = chunk_util.QueryFilter(callback)
+	if len(queries) == 0 {
```

Comment:
How can this happen?

Reply:
It should not happen, but I just added this to avoid any panics if, for some reason, we get zero queries, and we try to access the first element at line 78.